client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, Card } from '@automattic/components';
+import { getQueryArg } from '@wordpress/url';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { ReactElement, useCallback, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import FormRadio from 'calypso/components/forms/form-radio';
+import SearchCard from 'calypso/components/search-card';
+import { addQueryArgs } from 'calypso/lib/url';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, infoNotice } from 'calypso/state/notices/actions';
+import useAssignLicenseMutation from 'calypso/state/partner-portal/licenses/hooks/use-assign-license-mutation';
+import './style.scss';
+
+export default function AssignLicenseForm( { sites }: any ): ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ filter, setFilter ] = useState( false );
+	const [ selectedSite, setSelectedSite ] = useState( false );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const licenseKey = getQueryArg( window.location.href, 'key' ) as string;
+	const onSelectSite = ( site: any ) => setSelectedSite( site );
+
+	const siteCards = sites.map( ( site: any ) => {
+		if ( -1 !== site.domain.search( filter ) || false === filter ) {
+			return (
+				<Card key={ site.ID } className="assign-license-form__site-card">
+					<FormRadio
+						className="assign-license-form__site-card-radio"
+						label=""
+						name="site_select"
+						onClick={ () => onSelectSite( site.ID ) }
+					/>
+					{ site.domain }
+				</Card>
+			);
+		}
+	} );
+
+	const onSearch = ( query: any ) => setFilter( query );
+
+	const assignLicense = useAssignLicenseMutation( {
+		onSuccess: ( licenseKey: any ) => {
+			setIsSubmitting( false );
+			page.redirect( addQueryArgs( { highlight: licenseKey }, '/partner-portal/licenses' ) );
+		},
+		onError: ( error: Error ) => {
+			setIsSubmitting( false );
+			dispatch( errorNotice( error.message ) );
+		},
```

Comment:
**Remove initial notification/state**

AFAIK it's possible to pass an ID when dispatching a notice, so it can be programatically removed later (see `client/state/notices/actions.ts::removeNotice`).
Should we perhaps do that here, so we remove the initial `Assigning license...` message on errors?

![Screenshot 2022-02-09 at 13 06 10](https://user-images.githubusercontent.com/3846700/153198204-a6d3059f-a3b7-455a-accd-90bb25f5749e.png)

---

**Design differences**

Error messages appears differently than in the design - this is probably a way to limit the scope since we already have notifications, but just wanted to highlight it.

![Screenshot 2022-02-09 at 13 05 58](https://user-images.githubusercontent.com/3846700/153198195-3fd0732b-ab61-45b1-bd18-3b82a79a0dbd.png)

Reply:
I've removed the infoNotice in lieu of a different solution, so this is no longer an issue. As for the missing banner, we'll address this down the road, since right now we don't have a great way to map errors into the UI.